app/services/foreman_rh_cloud/url_remediations_retriever.rb

```diff
@@ -0,0 +1,33 @@
+module ForemanRhCloud
+  class UrlRemediationsRetriever < RemediationsRetriever
+    attr_reader :url, :payload, :headers
+
+    def initialize(url:, payload: '', headers: {}, logger: Logger.new(IO::NULL))
+      super(logger: logger)
+
+      @url = url
+      @payload = payload
+      @headers = headers
+
+      logger.debug("Querying playbook at: #{url} with payload: #{payload} and headers: #{headers}")
```

Comment:
Does this method actually perform the query? If not, I would think of moving this closer to the actual retrieval call.